tasks/provision_service.rb

```diff
@@ -7,174 +7,192 @@
-end
+# Provision and teardown vms through provision service.
+class ProvisionService
+  RETRY_COUNT = 3
 
-def platform_to_cloud_request_parameters(platform, cloud, region, zone)
-  case platform
-  when String
-    { cloud: cloud, region: region, zone: zone, images: [platform] }
-  when Array
-    { cloud: cloud, region: region, zone: zone, images: platform }
-  else
-    platform[:cloud] = cloud unless cloud.nil?
-    platform[:images] = [platform[:images]] if platform[:images].is_a?(String)
-    platform
-  end
-end
+  include PuppetLitmus::InventoryManipulation
 
-# curl -X POST https://facade-validation-6f3kfepqcq-ew.a.run.app/v1/provision --data @test_machines.json
-def invoke_cloud_request(params, uri, job_url, verb, retry_attempts)
-  headers =  {
-    'Accept' => 'application/json',
-    'Content-Type' => 'application/json'
-  }
-  headers['X-Honeycomb-Trace'] = ENV['HTTP_X_HONEYCOMB_TRACE'] if ENV['HTTP_X_HONEYCOMB_TRACE'] # legacy variable
-  headers['X-Honeycomb-Trace'] = ENV['HONEYCOMB_TRACE'] if ENV['HONEYCOMB_TRACE']
-
-  case verb.downcase
-  when 'post'
-    request = Net::HTTP::Post.new(uri, headers)
-    machines = []
-    machines << params
-    request.body = if job_url
-                     { url: job_url, VMs: machines }.to_json
-                   else
-                     { github_token: ENV['GITHUB_TOKEN'], VMs: machines }.to_json
-                   end
-  when 'delete'
-    request = Net::HTTP::Delete.new(uri, headers)
-    request.body = { uuid: params }.to_json
-  else
-    raise StandardError "Unknown verb: '#{verb}'"
+  def default_uri
+    'https://facade-release-6f3kfepqcq-ew.a.run.app/v1/provision'
   end
 
-  if job_url
-    File.open('request.json', 'wb') do |f|
-      f.write(request.body)
+  def platform_to_cloud_request_parameters(platform, cloud, region, zone)
+    case platform
+    when String
+      { cloud: cloud, region: region, zone: zone, images: [platform] }
+    when Array
+      { cloud: cloud, region: region, zone: zone, images: platform }
+    else
+      platform[:cloud] = cloud unless cloud.nil?
+      platform[:images] = [platform[:images]] if platform[:images].is_a?(String)
+      platform
     end
   end
 
-  req_options = {
-    use_ssl: uri.scheme == 'https',
-    read_timeout: 60 * 5, # timeout reads after 5 minutes - that's longer than the backend service would keep the request open
-    max_retries: retry_attempts # retry up to 5 times before throwing an error
-  }
+  # curl -X POST https://facade-validation-6f3kfepqcq-ew.a.run.app/v1/provision --data @test_machines.json
+  def invoke_cloud_request(params, uri, job_url, verb, retry_attempts)
+    headers =  {
+      'Accept' => 'application/json',
+      'Content-Type' => 'application/json'
+    }
+    headers['X-Honeycomb-Trace'] = ENV['HTTP_X_HONEYCOMB_TRACE'] if ENV['HTTP_X_HONEYCOMB_TRACE'] # legacy variable
+    headers['X-Honeycomb-Trace'] = ENV['HONEYCOMB_TRACE'] if ENV['HONEYCOMB_TRACE']
```

Comment:
we can remove all variables relating honeycomb from here now, it's no longer used